### DISCUSSION TITLE
User stories - brainstorming

### DISCUSSION THREAD

**Discussion by amis92** [BSData/phalanx#2] [Ideas]
2021-02-04 This list was reviewed and sorted as MVP and non-MVP at https://docs.google.com/document/d/1JBy8cduMbzUa38z3UTQcOOt4e-FBoJ12NItnK0LhOUg/edit?usp=sharing

---

Please share your ideas for user stories. Any and all are welcome.

- Things you do with your current roster editor and see as valuable
- Things you expect to be able to do with roster editor

## User Stories

These are a description of the functionality from the perspective of the end-user, and describes exactly what they want the system to do. It effectively states

> As a [type of user], I want [goal] so that [reason]

One benefit of user stories is that they do not require much technical knowledge to write.

**Comment by CloverFox** [comment]
as a user:

i would like to add repositories
so that i can load BSData into the application

i would like to update my repositories
so that i can have the most up to date data

i would like to download BSdata from the repository
so that i can use that data in the app

i would like to create a roster from the BSData
so that i can create my army list

i would like to save my roster
so that i can continue at a later date

i would like to delete old rosters
so that i can keep my list of rosters clean

i would like to name my roster
so that i can keep track of which roster is which

i would like to add units to my roster from the data
so that i can build up my army

i would like to add wargear options to the units
so that i can customise my units

i would like to view the army list
so that i can have a refernce to my selected options

**Comment by amis92** [comment]
as a user

I would like to see all possible rules for all possible options in the unit
so that I can have an overview of my choices in terms of rules

I would like to see what constraints are on my choices
so that I can plan changes without trying out the limits

I would like to search through available units and subselections by name
so that I don't have to look in all the different places they can be

**Comment by amis92** [comment]
as a user

I would like to easily report bugs
so that the data authors can fix them

as a data author

I would like to have bug reports that specify in detail where in the data the problem appears and possibly a repro roster
so that I can more easily fix issues

**Comment by amis92** [comment]
as a user

I would like to have an option of migrating my roster to new datafile set
so that I can keep my roster up-to-date with bug fixes and rule updates

I would like to have an option of saving a copy of the current roster before migrating to new datafile set
so that I can go back if somethings wrong in the new data version

I would like to see in detail what changes the migration resulted in
so that I can fix what the migration missed and be informed what did the updates change

**Comment by ResidentScribe** [comment]
As a user

I personally use the mobile app to play games, however I use the PC version to creat and experiment with lists.
I would like to see some improvements:

I would like to report bugs on the fly with ease.

Mobile preview needs some work, currently it is slow and sometimes doesn't work properly. PC preview can show individual weapons butt he app is limited to showing the whole unit.

RAM limiting, with older devices it chews through the stuff, crashing out of the app etc, I know android has an ability to limit RAM, not sure if its possible with apple.

**Comment by cartag** [comment]
As a data author

I would like it to verify "1 in 3/4/X" unit restrictions
so I can properly implement Age of Sigmar armies.

I would like a "battalion" level classification
So the roster editor will properly show what is a unit, instead of a collection of units rendered as a single unit. (Battalion -> Unit -> Model -> Wargear) 

As a user 

I would like to have a roster view and a "game" view
So I can quick reference my units in game.

I would l like a Favorite system
So I can easily save configured units.

I would like battalions to not show up as one giant lumped unit
So my roster will render units properly.

I would like to be able to duplicate configured units and battalions
So I can build lists with identical units easier

~~I would like to output my lists to PDF and a format I can import to aosreminders.com
So I can play my game easier with my created list.~~

I would like an output to sort abilities by phase
So I don't have to recreate my list or import on aosreminders.com

**Reply by cartag** [reply]
I had a person reach out to "fix" AoS output today, the battalion level designation would have made it a null issue in terms of output.

**Comment by amis92** [comment]
as a user

I would like to search my roster list
so that I can more easily find what I'm looking for

I would like to tag/categorize my rosters
so that I can browse my rosters more easily using filters

I would like to group rosters by game system
so that my armies from different games are separated

I would like to pin a couple of rosters on the top
so that I have quick access to my most important rosters

**Comment by amis92** [comment]
as a user

I would like to manage which data sources are used
so that I have visibility and control over them

**Comment by amis92** [comment]
as a data author
I would like for rule/profile content to be rendered as Markdown
so that users have see a pretty version of rule content

as a user
I would like to find all units that have a given rule/profile/category/weapon
so that I can see other sources/targets of the given item.

**Comment by OftKilted** [comment]
As a user:
I would like to be able to name my units
So I can 'better identify' my unique models (especially in Skirmish Games)

I would like to be able to do custom formatted to print my rosters (i.e. the .css customization for Killteam)
So I can make my printed rosters look customized like I want

As a Data Author
I want to be able to work with systems that do custom increments and decrements of stats
So I can better support campaign style games like Necromunda

**Comment by amis92** [comment]
as a user

I would like to undo/redo changes
so that I can more efficiently edit my rosters

I would like to have duplicate units collapsed and editable in batch or separately
so that multiples of the same unit are easily manageable and take less space on-screen

I would like to view a summary of what models do I need to field my army
so that I can prepare for games more easily

**Comment by cartag** [comment]
As a user 

I would like to be able to flag units as owned, not just as favorites, 
So I can filter when list building 

I would like to set army level selections similar to the AoS Warscroll Builder (drop downs) 
So I don't have to go searching for things like subfaction selection (a problem with AoS in mobile BS).

I would like customizable folders 
So I can organize my rosters

**Comment by amis92** [comment]
as a user

I would like to see what publications/books are being used in my roster
so that I can prepare better for games/tournaments

I would like to see from what publication is a given unit
so I can check myself for any errors or be informed of a new publication for that unit

I would like to see publication details
so I can find it in a store myself or use a link if available in details

**Comment by amis92** [comment]
as a data author

I would like to create extensions to existing datafiles
so that I can add fanmade units and community updates

as a user

I would like to use extension data sources
so that I can use community created data extensions/fanmade datafiles

**Comment by cartag** [comment]
As a data author 

I would like it to be system agnostic
So other titles can be supported

**Comment by cartag** [comment]
Not sure what scope we're planning on, so if these are out of line feel free to ignore

As a user 

I would like it to have the option to output as warscroll cards (with templates designed by data authors on a per game basis) 
So I can quick reference in game

As a Blood Bowl player

I would like to track independent statistics on units
So I can track experience, touchdowns, casualties, passes, etc.

**Reply by amis92** [reply]
I think we'll decide the scope after gathering suggestions. We'll start small for sure, but being aware of what might come in future is also important for setting the direction.

So go ahead, anything is perfectly fine at this stage 😅

**Comment by GuillermoDiazGarcia** [comment]
As a user

 I would like to copy elements x number of times instead of having to click every time, which is tedious in a lot of contexts

I would like things to process on the device itself to avoid seeing infinite loading screens if my internet connection is shaky

**Comment by Price-VanSaint** [comment]
as a user:

This is pure wish-listing and no idea how well it might play together with each other

I would like the output format to be collapsible and expand certain sections with a tap  (Elites, HQ etc.)
either in a tabbed browser alike or a tiles layout that can be dragged and dropped

I would like to filter to keywords  like <CORE>  or highlight tiles/tabs that have keywords that are applicable to some stratagems

I would like rules to be hidden or shown with a change to default settings and shown as popup maybe when tapped

when making lists

 I'd like filters for keywords or a search

I'd like easy duplication of units or models with specific loadouts (Go silver tide  !)  or the ability to save a squad (gauss immortals instead of tesla immortals etc.)   i see this as valuable for units that have many wargear options on a per model basis  (Tomb blades)

I'd like the easy ability to drag in a forgeworld or other unit from different codex  (Say chaos into deathguard etc.) without nesting detachments or having to load the entire new codex.

I'd like validation errors to be linked so tapping the error goes to the relevant unit / error to be fixed.

I'd like a pony

I'd like to be able to adjust points limits and detachments while in list and not have to exit out

I'd like fortifications and unaligned or Superheavy aux etc.  to be highlighted so they can see what detachment rules or other things do and do not apply to them  (yeah go on wave that magic wand!)

I'd like to go large on that, and finally I'd like to thank The Academy and my Mom !

Good luck y'all

**Comment by amis92** [comment]
as a user

I would like to be able to add my own/modify existing units and rules and save them for future use and have them work with updated data
So that I can fill in gaps, fix errors and add custom house rules on my own

**Comment by LIBEDRAT0R** [comment]
As a user I would like to be able to create my own units without having to redo them when data updates.

Also, I would love to have the supa skorcha and skorcha options for the wartrak and kombi weapons for the nob bikers as legends option. 

Thanks in advance.

**Comment by cartag** [comment]
As a data author

I would like to set Category requirements for battalions
So choices for "0-5 ZOMBIE" units would list anything categorized as a Zombie

**Comment by ghost** [comment]
As a data files developer, I would like to be able to pop up messages at users before they update their files to share changelists, breaking notices, etc. This way they won't find their lists blown up blindly when they update data.

**Comment by ghost** [comment]
As a data editor, I would like to able to nest entity descriptions (like selection groups) under taxonomy entries (like categories).

**Comment by cartag** [comment]
As a desktop user

I would like it to show the item I'm hovering over in a tooltip
So I can easily see what something does

**Comment by ghost** [comment]
As a data files editor, I would like not to have to create entries and links except when absolutely required, and I should be able to promote a entry to a shared entry with the click of a button and no links breaking.

**Comment by Manethak** [comment]
as a user I would like to be able to share my lists with people

as a user I would like to be shown when my list has validation errors

as a data developer I would like the system to support many different game systems

as a developer I would like a way to integrate with other applications (probably an API)

**Comment by cartag** [comment]
As a data author

I would like it to check file versions when updating data
So anything I'm working on but haven't uploaded yet doesn't get erased

**Comment by amis92** [comment]
This list was reviewed and sorted as MVP and non-MVP at https://docs.google.com/document/d/1JBy8cduMbzUa38z3UTQcOOt4e-FBoJ12NItnK0LhOUg/edit?usp=sharing

**Comment by Tigerwraith** [comment]
As a data author, I want the different sections to be scaleable so that if the game system being entered only needs 2 sections it shows only 2 areas,bid you need 4 then you have 4 on PC, on mobile you could make the different sections to be swipeable.

**Comment by Mayegelt** [comment]
I hope you are sitting comfortably because it's story storm.

So in the Horus Heresy files I was the one who added some of the player made army lists and modifications.
So I started with the Eldar and Necron stuff then did the Mournival changes to the big Legiones Astartes file.
To those who are unaware the Mournival changes are, basically just a few modifications to the core rules and selections.

Most of the Mournival stuff is in its own cat file which reference the main file and are mainly hidden. But there are a few options that needed to be put in the main LA file. But it wasn't actually that much.
Originally I had when I first did the Eldar one I just hosted it myself with instructions how to add it to the files. Because it rarely needed an update, once added it would be fine.
Issues were that people struggled to still put it in. Especially iOS users.

Anyway... Little busy ATM. Will check on this again and add a bunch of features I'd like to see.

**Comment by cartag** [comment]
As a data author

I would like Modifiers on Constraints on Force Entries to be enforced
So I can set points limits independent of what they are set to when the roster is created. 
- Context: The below pic should be showing max die rolls of 5 as an alert
![image](https://user-images.githubusercontent.com/4388344/107863967-45d6b680-6e15-11eb-9a45-3fec7b1b02e6.png)

I would like to be able to hide and unhide Cost Types, Category Entries, Force Entries
So I can take advantage of features that we are capable of setting but aren't implemented properly now. 

I would like to set hidden Category Entries to make data authoring easier
So I won't get bug reports when units have categories that are only used on the backend that show up in roster printout

**Comment by cartag** [comment]
As a data author

I would like to change primary category on SSE without doubling happening when it changes

**Comment by CarlHeiberg** [comment]
Enable customizations of roster creation dropdowns

- Names of dropdowns
- Number of dropdowns
- Values in dropdowns
- Semantic meaning of catalogs and forces

**Reply by cartag** [reply]
Don't mock my poor photoshop skills. 

The ability to do drop downs like in the Leader Essence section, with a list of choices for either single selection, or check boxes for multiple, would be a great UI upgrade, or at least option, for data authors. 

![BS Mockup](https://user-images.githubusercontent.com/4388344/187008748-522411a6-2589-4a5c-97dd-6712575f3a00.png)

**Comment by pantheonw** [comment]
As a user, I want the option to view my roster with permanent modifications to profiles applied so that the changes aren't something I need to process when checking characteristics.
